### PR TITLE
style-guide: organize content and update sections; *: fix Markdownlint errors

### LINKED ISSUE #12008: Organize style guide content
It would be nice to split the content there in some sections and add a contents subtitle in the start.

Here is my proposal:
```md
## Contents

1. [General layout](#generallayout)
2. [General rules](#generalrules)
3. [Special pages](#specialpages)
4. [Heading](#heading)
5. [Example descriptions](#exdescriptions)
6. [Example commands](#excommands)
7. [Language-specific rules](#langrules)
```

### PR DESCRIPTION
**Pull request by vitorhcl**
- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).

## Detailed Changelog

- This PR rearranges all the style guide sections to follow the same flow as a tldr page (for more details, checkout #12008)
- Added concise information about keycaps and their syntax to the style guide.
- Added documentation about string placeholders and the preference of "{{placeholder}}" instead of {{"placeholder"}}.
- Added documentation about optional placeholders (with a suggestion to use the description instead of any special syntax)
- Added a note to the top of the contributing file to suggest new users/translators to checkout the style guide and the translation template.
- Renamed `.markdownlintrc` to the currently used `.markdownlint.json` file. (As till now, the checks weren't running properly due to the unsupported file name)
- Disabled Markdown lint rule "MD029" (to allow mixing and matching ordered numbered list and unordered list in language-specific rules sections); for more information about this rule checkout [this](https://github.com/DavidAnson/markdownlint/blob/main/doc/md029.md) file.
- Added markdown lint disable parameters for client specification to its header for [MD007](https://github.com/DavidAnson/markdownlint/blob/main/doc/md007.md), [MD013](https://github.com/DavidAnson/markdownlint/blob/main/doc/md013.md) and [MD024](https://github.com/DavidAnson/markdownlint/blob/main/doc/md024.md). [If we want we can optionally, fix the issues and remove the disable parameters in the future]
- Added markdown lint disable parameters for README and LICENSE file for the rule [MD041](https://github.com/DavidAnson/markdownlint/blob/main/doc/md041.md). [If we want we can optionally, fix the issues and remove the disable parameters in the future]
- Fixed markdownlint and minor issues in the CONTRIBUTING file.
- Fixed markdownlint issues in COMMUNITY-ROLES and git-terminal file.

Closes #12008, #12396, #12628.

### CODE REVIEW

**Review by kbdharun — COMMENTED** [review]
Great changes, sorry for the delay. I will get back to this PR soon.

**Comment by sbrl** [comment]
Hmmm, I would suggest that the manual contents isn't particularly useful given as you say GH provides an automatic one. It would also require manually updating every time we add/remove/rename sections.

Having said this, it does organise the guide nicely and make what appears to be an unstructured document much more grokable, so I think it's worth the effort of keeping up to date - especially since it doesn't contain second-level items.

**Review by sbrl — COMMENTED** [review]
Thanks for the PR, @vitorhcl! Organising the style guide definitely needs doing, and this should make it more understandable for new users. Great job!

I've left some comments below.

File: contributing-guides/style-guide.md
Comment (vitorhcl) [reply contributing-guides/style-guide.md:112]: I didn't edit this text, I just separated it from the "Imperative Mood" section because it didn't fit there, it almost looked like a random text in the middle of the section.

File: contributing-guides/style-guide.md
Comment (vitorhcl) [reply contributing-guides/style-guide.md:106]: From the HTML spec of 15 February 2024, section [7.4.6.4 Scrolling to a fragment](https://html.spec.whatwg.org/multipage/browsing-the-web.html#scrolling-to-a-fragment):
> To find a potential indicated element given a [Document](https://html.spec.whatwg.org/multipage/dom.html#document) document and a string fragment, run these steps:
> 
> If there is an element [in the document tree](https://dom.spec.whatwg.org/#in-a-document-tree) whose [root](https://dom.spec.whatwg.org/#concept-tree-root) is document and that has an [ID](https://dom.spec.whatwg.org/#concept-id) equal to fragment, then return the first such element in [tree order](https://dom.spec.whatwg.org/#concept-tree-order).

>
> If there is an [a](https://html.spec.whatwg.org/multipage/text-level-semantics.html#the-a-element) element [in the document tree](https://dom.spec.whatwg.org/#in-a-document-tree) whose [root](https://dom.spec.whatwg.org/#concept-tree-root) is document that has a [name](https://html.spec.whatwg.org/multipage/obsolete.html#attr-a-name) attribute whose value is equal to fragment, then return the first such element in [tree order](https://dom.spec.whatwg.org/#concept-tree-order).
>
> Return null.

So either would work, but I think it's better to use `id`

**Comment by vitorhcl** [comment]
I think we can use a tip box instead of updating a table of contents, given that the GitHub one list subheaders.

What do you think @sbrl @sebastiaanspeck @kbdharun?

**Comment by kbdharun** [comment]
> I think we can use a tip box instead of updating a table of contents, given that the GitHub one list subheaders.

Yeah, that would work too. Btw, can we get back to this PR once #12304 is addressed/merged to prevent merge conflicts?

File: contributing-guides/style-guide.md
Comment (sbrl) [reply contributing-guides/style-guide.md:112]: Okay, I see. But considering your PR in general the diff is quite extensive, is it just table of content headings? The diff has gone a bit mad.

**Review by kbdharun — COMMENTED** [review]
> > > I think we can use a tip box instead of updating a table of contents, given that the GitHub one list subheaders.
> > 
> > 
> > Yeah, that would work too. Btw, can we get back to this PR once #12304 is addressed/merged to prevent merge conflicts?
> 
> #12304 has been merged, but there are conflicts which @vitorhcl needs to resolve

I will fix them and make other changes (I have some free time to work on this today).

---

I would personally suggest not to mix and match HTML tags in Markdown files as it would trigger Markdown lint errors/warnings (we could set it to be ignored but it isn't suggested to have these tags as some Git providers don't support rendering HTML in Markdown pages); instead we can use the same tags as the page names. I will convert them back again shortly.

**Comment by vitorhcl** [comment]
> I would personally suggest not to mix and match HTML tags in Markdown files as it would trigger Markdown lint errors/warnings (we could set it to be ignored but it isn't suggested to have these tags as some Git providers don't support rendering HTML in Markdown pages)

If we are also targeting other Git providers, my suggestion is to use a table of contents just with just 1 level of depth, since GitHub is the only provider that has an automatic TOC in their UI TTBOMK.

Also, I think it would be easy to mantain it because I don't think the base structure will be commonly changed.

**Comment by vitorhcl** [comment]
Hi @kbdharun, just sending a friendly nudge on this important PR :)

File: contributing-guides/style-guide.md
Comment (kbdharun) [reply contributing-guides/style-guide.md]: I have reverted the HTML tag addition to titles and added the index reference to the markdown headings directly.

File: contributing-guides/style-guide.md
Comment (kbdharun) [reply contributing-guides/style-guide.md:112]: I have currently rebased this PR to accommodate the recent changes to the style guide, I will provide a detailed changelog once my changes are complete.

**Review by kbdharun — APPROVED** [review]
LGTM, thanks for your contribution.

----

I am done with my changes and fixes, @sbrl, @sebastiaanspeck and @vitorhcl I have added a detailed changelog of @vitorhcl's and my changes to the PRs description. Feel free to check it out.

**Comment by kbdharun** [comment]
Seems like I need to manually rebase the contents of https://github.com/tldr-pages/tldr/commit/5da4cacd9c1f40e47961e67dacb17f7ed024e6c5 and https://github.com/tldr-pages/tldr/commit/21ad18714bf63cde74f10fbb82c20f454da93f5a to the current version, will do it later today.

**Review by gutjuri — APPROVED** [review]

File: contributing-guides/style-guide.md
Comment (vitorhcl) [inline contributing-guides/style-guide.md:443]:
```suggestion
  For instance, if multiple paths are expected `{{path/to/directory1 path/to/directory2 ...}}` can be used.
```

```suggestion
  For instance, if multiple paths are expected, use `{{path/to/directory1 path/to/directory2 ...}}`.
```

File: contributing-guides/style-guide.md
Comment (vitorhcl) [inline contributing-guides/style-guide.md:399]:
```suggestion
> It is suggested to enclose placeholders accepting strings as input within quotes. i.e. Use `"{{placeholder}}"` instead of `{{"placeholder"}}`.
```
IMO here we can make it a bit open for edge cases.

**Review by vitorhcl — COMMENTED** [review]
Awesome, thank you for your extensive cleanup on this, @kbdharun!

LGTM after the conversations opened by me are resolved.

File: contributing-guides/style-guide.md
Comment (kbdharun) [reply contributing-guides/style-guide.md:288]: How about this?

```suggestion
- We suggest using the more information link of the English page for translations.
```
(or)
```suggestion
- It is suggested to use the more information link of the English page for translations.
```

We don't generally support showing localized versions of the pages in a majority of cases as the translations a lot of times are out of sync with the English page.

File: contributing-guides/style-guide.md
Comment (vitorhcl) [inline contributing-guides/style-guide.md:288]: > We don't generally support showing localized versions of the pages in a majority of cases as the translations a lot of times are out of sync with the English page.

My understanding: it's because the links can be outdated if they change, right?

If so, I agree this can be a problem. I prefer "It is suggested to", but I support making the argument clear in the style guide itself:

```suggestion
- It is suggested to use a more information link with English content in both translations and English pages. That's because the links can eventually change, but the translations are often out of sync with the English pages.
```

I've also thought that a link update can eventually come from a translation and propagate to the others using our script, so I reworded the first sentence, what do you think?

**Review by kbdharun — APPROVED** [review]

File: contributing-guides/style-guide.md
Comment (gutjuri) [reply contributing-guides/style-guide.md]: I'd personally prefer the first version.

**Review by gutjuri — APPROVED** [review]

**Pull request merged**
